test/box/misc.test.lua

```diff
@@ -363,3 +363,14 @@ i.parts
 i = s:create_index('test6', {parts = {4, 'string'}})
 i.parts
 s:drop()
+
+--
+-- gh-5473: forbid specifying index options in key parts
+--
+--
+s = box.schema.space.create('test')
+i = s:create_index('test1', {parts = {1, 'unsigned', unique=false}})
+i = s:create_index('test1', {parts = {1, 'unsigned', distance=3}})
+i = s:create_index('test1', {parts = {2, 'string', 3, 'string', unique=false}})
+i = s:create_index('test1', {parts = {2, 'string', 3, 'string', distance=3}})
```

Comment:
I'd also add test case with several separate key parts:
```
s:create_index('test1', {parts = {{1,'int', distance=3}, {field=2, type='int'}}})
s:create_index('test1', {parts = {{1,'int'}, {field=2, type='int', type = 'hash'}}})

```
Note that unnamed entries are still ignored:
```
box.space.tester:create_index('sc3', {parts = {{1,'int'}, {field=2, type='int', 'asd'}}})
- unique: true
  parts:
  - type: integer
    is_nullable: false
    fieldno: 1
  - type: integer
    is_nullable: false
    fieldno: 2
  hint: true
  id: 2
  type: TREE
  space_id: 512
  name: sc3
```
Meanwhile in single-array input they are validated:
```
s:create_index('sc4', {parts = {1,'int', 'asd'}})
- error: 'Illegal parameters, options.parts: expected field_no (number), type (string)
    pairs'
...

```

Reply:
Fixed